### PR TITLE
Extension metadata generation improvements

### PR DESCRIPTION
**Pull request by mathewc**
Addresses https://github.com/Azure/azure-functions-host/issues/3186. Supporting SDK changes and discusion here: https://github.com/Azure/azure-webjobs-sdk/pull/1829.

Summary: Now, rather than generating:

```javascript
{
  "extensions":[
    { "name": "EventHubConfiguration", "typeName":"Microsoft.Azure.WebJobs.ServiceBus.EventHubConfiguration, Microsoft.Azure.WebJobs.EventHubs, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null"},
    { "name": "ServiceBusExtensionConfig", "typeName":"Microsoft.Azure.WebJobs.ServiceBus.Config.ServiceBusExtensionConfig, Microsoft.Azure.WebJobs.ServiceBus, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null"}
  ]
}
```

we'll generate:

```javascript
{
  "extensions":[
    { "name": "EventHubs", "typeName":"Microsoft.Azure.WebJobs.EventHubs.EventHubsWebJobsStartup, Microsoft.Azure.WebJobs.EventHubs, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null"},
    { "name": "ServiceBus", "typeName":"Microsoft.Azure.WebJobs.ServiceBus.ServiceBusWebJobsStartup, Microsoft.Azure.WebJobs.ServiceBus, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null"}
  ]
}
```

And the indicated startup classes will be invoked by the runtime to register the extensions.

### CODE REVIEW

File: tools/ExtensionsMetadataGenerator/test/ExtensionsMetadataGeneratorTests/ExtensionsMetadataGeneratorTests.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿using System.Linq;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace ExtensionsMetadataGeneratorTests
+{
+    public class ExtensionsMetadataGeneratorTests
+    {
+        [Fact]
+        public void GenerateExtensionReferences_Succeeds()
```
Comment (mathewc) [inline tools/ExtensionsMetadataGenerator/test/ExtensionsMetadataGeneratorTests/ExtensionsMetadataGeneratorTests.cs]: These are tests. They test things. Tests are good. All our code should have tests :)

File: tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator.Console/ExtensionsMetadataGenerator.cs [hunk 2]
```diff
@@ -16,6 +15,8 @@ namespace ExtensionsMetadataGenerator
 {
     public class ExtensionsMetadataGenerator
```
[the remark this answers is not in the document]
Comment (mathewc) [reply tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator.Console/ExtensionsMetadataGenerator.cs]: Update what namespace?

File: tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator.Console/ExtensionsMetadataGenerator.Console.csproj [hunk 3]
```diff
@@ -1,8 +1,8 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
```
[the remark this answers is not in the document]
Comment (mathewc) [reply tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator.Console/ExtensionsMetadataGenerator.Console.csproj]: Fixed

File: tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator.Console/ExtensionsMetadataGenerator.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (fabiocav) [reply tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator.Console/ExtensionsMetadataGenerator.cs]: From ExtensionsMetadataGenerator so a proper namespace. That also means we’d need to update the target file

[the remark this answers is not in the document]
Comment (mathewc) [reply tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator.Console/ExtensionsMetadataGenerator.cs]: I'll do this in a follow up

**Pull request closed without merging**